easybuild/framework/easyconfig/exts_tools.py

```diff
@@ -342,6 +348,268 @@ def _format_metadata_as_extension(pkg_class, pkg_metadata, bioconductor_version=
+            if is_in_exts_list:
+                continue
+
+            # check if the dependency is in the exclude list
+            is_excluded = False
+            for exclude_ext in EXCLUDE_R_LIST:
+                if exclude_ext.lower() == dep_name.lower():
+                    is_excluded = True
+                    print_msg(f"\t{dep_name:<{PKG_NAME_OFFSET}} is blacklisted", prefix=False, log=_log)
+                    continue
+
+            # if the dependency is excluded, then skip
+            if is_excluded:
+                continue
+
+            # check if the dependency is already installed by a dependency
+            is_installed = False
+            for inst_ext in installed_exts:
+                inst_ext_name, _, inst_ext_options = _get_extension_values(inst_ext)
+                if inst_ext_name.lower() == dep_name.lower():
+                    is_installed = True
+                    print_msg(
+                        f"\t{dep_name:<{PKG_NAME_OFFSET}} installed by dependency: {inst_ext_options['easyconfig_path']}", prefix=False, log=_log)
+                    break
+
+            # if the dependency is already installed, then skip
+            if is_installed:
+                continue
+
+            print_msg(f"\t{dep_name:<{PKG_NAME_OFFSET}} added as dependency", prefix=False, log=_log)
+            
+            # build the metadata dependency as extension getting the last version
+            dep_name = {'name': dep_name, 'version': None, 'options': {}}
+
+            # recursively get dependencies of dependency
+            deps = _get_R_extension_dependencies(dep_name, bioconductor_version, processed_exts)
+
+            # append the dependencies to the list
+            dependencies.extend(deps)
+
+            # append the dependency to the list
+            dependencies.append(dep_name)
+
+    return dependencies
+
+
+def _get_completed_R_exts_list(exts_list, bioconductor_version=None, installed_exts=[]):
+    """
+    Complete the R extensions list with its dependencies in correct order.
+
+    :param exts_list: list of extensions to be updated.
+    :param bioconductor_version: bioconductor's version to use (if any)
+    :param installed_exts: list of installed extensions by dependencies (if any)
+
+    :return: list of extensions for a complete exts_list
+    """
+
+    # check if the exts_list is empty
+    if not exts_list:
+        raise EasyBuildError("No exts_list provided for completing")
+
+    # get the dependendy tree. i.e. list of dependencies for each extension
+    dependendy_tree = []
+    for ext in exts_list:
+
+        # get the values of the extension
+        ext_name, ext_version, ext_options = _get_extension_values(ext)
+
+        print()
+        print_msg("Dependencies of '%s':" % ext_name, log=_log)
+
+        # get dependencies of the extension
+        dependencies = _get_R_extension_dependencies(ext, bioconductor_version, exts_list, installed_exts)
```

Comment:
As it is defined, you can get a multiple coincidences for a dependency if it appears in several packages. I guess that it is already covered in EasyUpdate.